pkg/operator/controller/ingress/controller.go

```diff
@@ -548,6 +548,7 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 					changed = true
 				}
 				if statusLB.ProviderParameters.AWS.Type == operatorv1.AWSClassicLoadBalancer {
+					statusLB.ProviderParameters.AWS.NetworkLoadBalancerParameters = nil
```

Comment:
statusLB.ProviderParameters.AWS.NetworkLoadBalancerParameters   does not the .NetworkLoadBalancerParameters give a panic error for nil  as it will access NetworkLoadBalancerParameters which does not exist?

Or may be we need to add a condition to check statusLB.ProviderParameters.AWS.NetworkLoadBalancerParameters for nil.

Reply:
No need to worry about a nil pointer dereference when assigning nil to `statusLB.ProviderParameters.AWS.NetworkLoadBalancerParameters`. The [logic](https://github.com/openshift/cluster-ingress-operator/blob/7edef371026129f765a86748148c79b38fba636e/pkg/operator/controller/ingress/controller.go#L543-L545) above this PR's changes already ensures that `statusLB.ProviderParameters.AWS` is non-nil, so this assignment is safe. Assigning nil to this field is simply setting a pointer to nil (doesn't matter if it's already nil) and doesn’t cause a panic as long as the parent structure exists.